aea/protocols/generator/generator.py

```diff
@@ -2405,50 +2012,61 @@ def _generate_file(self, file_name: str, file_content: str) -> None:
+        Run the generator. If in "full" mode (protobuf_only is False), it:
+        a) validates the protocol specification.
+        b) creates the protocol buffer schema file.
+        c) generates python modules.
+        d) applies black formatting
+
+        If in "protobuf only" mode (protobuf_only is True), it only does a) and b).
 
+        :param protobuf_only: mode of running the generator.
         :return: None
         """
         # Create the output folder
         output_folder = Path(self.output_folder_path)
         if not output_folder.exists():
             os.mkdir(output_folder)
 
-        # Generate the protocol files
-        self._generate_file(INIT_FILE_NAME, self._init_str())
-        self._generate_file(PROTOCOL_YAML_FILE_NAME, self._protocol_yaml_str())
-        self._generate_file(MESSAGE_DOT_PY_FILE_NAME, self._message_class_str())
-        if (
-            self.protocol_specification.dialogue_config is not None
-            and self.protocol_specification.dialogue_config != {}
-        ):
-            self._generate_file(DIALOGUE_DOT_PY_FILE_NAME, self._dialogue_class_str())
-        if len(self._all_custom_types) > 0:
-            self._generate_file(
-                CUSTOM_TYPES_DOT_PY_FILE_NAME, self._custom_types_module_str()
-            )
-        self._generate_file(
-            SERIALIZATION_DOT_PY_FILE_NAME, self._serialization_class_str()
-        )
         self._generate_file(
             "{}.proto".format(self.protocol_specification.name),
             self._protocol_buffer_schema_str(),
         )
 
-        # Warn if specification has custom types
-        if len(self._all_custom_types) > 0:
-            incomplete_generation_warning_msg = "The generated protocol is incomplete, because the protocol specification contains the following custom types: {}. Update the generated '{}' file with the appropriate implementations of these custom types.".format(
-                self._all_custom_types, CUSTOM_TYPES_DOT_PY_FILE_NAME
+        if not protobuf_only:
+            # Generate Python files
+            self._generate_file(INIT_FILE_NAME, self._init_str())
+            self._generate_file(PROTOCOL_YAML_FILE_NAME, self._protocol_yaml_str())
+            self._generate_file(MESSAGE_DOT_PY_FILE_NAME, self._message_class_str())
+            if (
+                self.protocol_specification.dialogue_config is not None
+                and self.protocol_specification.dialogue_config != {}
+            ):
+                self._generate_file(
+                    DIALOGUE_DOT_PY_FILE_NAME, self._dialogue_class_str()
+                )
+            if len(self.spec.all_custom_types) > 0:
+                self._generate_file(
+                    CUSTOM_TYPES_DOT_PY_FILE_NAME, self._custom_types_module_str()
+                )
+            self._generate_file(
+                SERIALIZATION_DOT_PY_FILE_NAME, self._serialization_class_str()
             )
```

Comment:
This should be wrapped in a public method `generate_python_files` so it becomes a single line here

Reply:
Done